core/src/commonMain/kotlin/com/ustadmobile/staging/core/view/SchoolEditView.kt

```diff
@@ -0,0 +1,30 @@
+package com.ustadmobile.staging.core.view
+
+import com.ustadmobile.core.view.UstadView
+import com.ustadmobile.lib.db.entities.School
+
+/**
+ * Core View. Screen is for SchoolDetail's View
+ */
+interface SchoolEditView : UstadView {
+
+    /**
+     * Method to finish the screen / view.
+     */
+    fun finish()
+
+    fun setSchool(school: School)
+
+    fun setPicture(picturePath: String)
+
+    //Add any other view bits here..
+
+    companion object {
+        // This defines the view name that is an argument value in the go() in impl.
+        const val VIEW_NAME = "SchoolDetail"
+        const val ARG_SCHOOL_DETAIL_SCHOOL_UID = "ArgSchoolDetailSchoolUid"
+        const val ARG_SCHOOL_NEW = "ArgSchoolNew"
```

Comment:
Why does the edit presenter need this argument for whether or not a school is new? If it is new, then no uid is provided.